index.html

```diff
@@ -380,16 +380,498 @@
 ]
 
 ---
-
 class: center, middle
 
-# Redux: qu'est-ce qui change ?
+# Et Redux dans tout ca ?
+--
+## et puis d'abord, _pourquoi_ Redux?
+---
+# .smaller[La gestion de l'etat d'une application]
+### Avant React:
+--
+.smaller[L'etat est dissemine entre le DOM, des variables globales, Local Storage, etc...]
+
+.smaller[
+```javascript
+*  var path = window.location.hash;
+*  var $tab = $(`.tab[href="#${path}"]`);
+   var shouldBeActive = !$($tab).hasClass('is-active');
+   if (shouldBeActive) {
+      $('.tab.is-active').removeClass('is-active');
+*     $tab.addClass('is-active');
+   }
+```
+]
+---
+# .smaller[La gestion de l'etat d'une application]
+### Avec React:
+
+.smaller[L'etat est difficile a propager a d'autres composants.]
+.smallest[
+```javascript
+  class Tabs extends React.PureComponent {
+    state = { activeTab: 'businesses' }
+
+    handleClick = (evt) => {
+       const newActiveTab = evt.currentTarget.href.split('#')[1];
+*      this.setState(
+*        (prevState) => prevState.activeTab === newActiveTab
+*        ? prevState
+*        : { activeTab: newActiveTab }
+      );
+    }
+
+    render() {
+      return (
+        <ul>
+          {tabs.map((tab) => (
+            <li key={tab-${this.props.tab.name}}>
+*             <a onClick={this.handleClick}
+*                className={this.state.activeTab === ? tab.name : ''}
+                 href={`#${this.props.tab.name}`}
+              >
+                {this.props.tab.label}
+              </a>
+            </li>
+          ))}
+        </ul>
+      );
+    }
+  }
+```
+]
+---
+# .smaller[La gestion de l'etat d'une application]
+### Avec React:
+
+_“Lift the state UP”_ pattern to the rescue?
+.smallest[
+```javascript
+  class App extends React.PureComponent {
+    tabs = [{ name: 'individuals' }, { name: 'businesses' }, { name: 'developers' }],
```

Comment:
je suis pas sûr que la virgule à la fin soit légale; soit on met rien soit un point-virgule, je pense

Reply:
`;` it is, fixed